base_layer/mmr/src/balanced_binary_merkle_proof.rs

```diff
@@ -81,6 +94,8 @@ pub enum MergedBalancedBinaryMerkleProofError {
     CantMergeZeroProofs,
     #[error("Bad proof semantics")]
     BadProofSemantics,
+    #[error("Math overflow")]
+    MathOverflow,
```

Comment:
Perhaps consider renaming the entire `enum` to something like `BalancedBinaryMerkleProofError`, since right now a non-merged proof can return a merged proof error.